apps/framework-cli/src/cli/local_webserver.rs

```diff
@@ -1290,30 +1289,109 @@ fn handle_listener_err(port: u16, e: std::io::Error) -> ! {
         _ => panic!("Failed to listen to port {}: {:?}", port, e),
     }
 }
-async fn shutdown(settings: &Settings, project: &Project, graceful: GracefulShutdown) -> ! {
+async fn shutdown(
+    settings: &Settings,
+    project: &Project,
+    graceful: GracefulShutdown,
+    redis_client: Arc<Mutex<RedisClient>>,
+) -> ! {
+    // First, initiate the graceful shutdown of HTTP connections
+    let shutdown_future = graceful.shutdown();
+
+    // Wait for connections to close with a timeout
     tokio::select! {
-            _ = graceful.shutdown() => {
-                info!("all connections gracefully closed");
-            },
-            _ = tokio::time::sleep(std::time::Duration::from_secs(10)) => {
-                warn!("timed out wait for all connections to close");
+        _ = shutdown_future => {
+            info!("all connections gracefully closed");
+        },
+        _ = tokio::time::sleep(std::time::Duration::from_secs(10)) => {
+            warn!("timed out wait for all connections to close");
         }
     }
 
+    // Ensure any Redis-using tasks are stopped
+    info!("Stopping any tasks that might be using Redis");
+    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+
+    // Shutdown Redis client that was passed
+    info!("Shutting down Redis client connections");
+
+    // First attempt: Try to properly shut down the Redis client
+    match tokio::time::timeout(std::time::Duration::from_secs(3), async {
+        if let Ok(mut client) = redis_client.try_lock() {
+            match client.shutdown().await {
+                Ok(_) => info!("Redis client shutdown completed successfully"),
+                Err(err) => warn!("Error during Redis client shutdown: {:?}", err),
+            }
+        } else {
+            warn!("Could not acquire lock on Redis client for shutdown");
+        }
+    })
+    .await
+    {
+        Ok(_) => info!("Redis shutdown completed within timeout"),
+        Err(_) => warn!("Redis shutdown timed out, proceeding with application shutdown"),
+    }
+
+    // Important: Drop our reference to the Redis client to ensure it's fully cleaned up
+    // before we start stopping containers
+    info!("Dropping Redis client reference");
+    drop(redis_client);
```

Comment:
Why is that need to be done manually if the client is shutdown?

Reply:
In this comment I assume you were referring to the sleep operation which like the earlier one can be safely removed.  The drop shown above is necessary.